Dockerfile

```diff
@@ -1,20 +1,23 @@
 FROM aeternity/builder:bionic-otp23 as builder
 
+# Add rebar configuration and fetch dependencies
+ADD ./Makefile ./rebar* /app/
+RUN cd /app && make get-deps
 # Add the whole project and compile it.
 ADD . /app
 RUN cd /app && make prod-build
```

Comment:
Since `ADD . /app` is unlikely to fail, I don't understand why you are doing `ADD ./Makefile ./rebar* /app/` in a separate step before `RUN cd /app && make get-deps` when there is a `ADD . /app` after that.

If you want to fetch dependencies in a separate step for better caching you can do this instead:

```suggestion
# Add the whole project and compile it.
ADD . /app
RUN cd /app && make get-deps
RUN cd /app && make prod-build
```

Reply:
It is a tricky part. Doing `ADD . /app` before `make get-deps` then dependencies will be installed again if anything changes. If I add only files that declares dependencies then the installed dependencies would be cached by docker and won't be installed again if the source code changes.

The motivation is that I was working from a mobile network and would like to save some traffic without setting up the local erlang environment. In general, setup caching for the dependencies layer is a good practice, here is the same for nodejs: https://docs.docker.com/language/nodejs/build-images/#create-a-dockerfile-for-nodejs